docs/topics/db/aggregation.txt

```diff
@@ -632,3 +634,36 @@ aggregate that author count, referencing the annotation field:
     >>> from django.db.models import Avg, Count
     >>> Book.objects.annotate(num_authors=Count("authors")).aggregate(Avg("num_authors"))
     {'num_authors__avg': 1.66}
+
+Handling empty querysets and groups using aggregate ``default``
+---------------------------------------------------------------
+
+When an aggregate function is used with an empty ``QuerySet`` or grouping,
+``None`` is returned as a default. This is because aggregate functions result
+in ``NULL`` when used with an empty result set. The optional ``default``
+argument for aggregate functions allows specifying a value to use as a
+default when the executed query returns no rows. Some aggregate functions
+do not support the ``default`` argument because they already handle this, e.g.
+``Count`` returns ``0``.
```

Comment:
I have re-written this paragraph trying to reduce some of the duplication. I'll push a new revision soon.